picker.go

```diff
@@ -0,0 +1,56 @@
+package grpc
+
+import (
+	"time"
+)
+
+// Picker picks a Conn for RPC requests.
+// This is EXPERIMENTAL and Please do not implement your own Picker for now.
+type Picker interface {
+	// Pick returns the Conn to use for the upcoming RPC. It may return different
+	// Conn's up to the implementation.
+	Pick() (*Conn, error)
+	// State returns the connectivity state of the underlying connections.
+	State() ConnectivityState
+	// WaitForStateChange blocks until the state changes to something other than
+	// the sourceState or timeout fires on cc. It returns false if timeout fires,
+	// and true otherwise.
+	WaitForStateChange(timeout time.Duration, sourceState ConnectivityState) bool
```

Comment:
This is a loose end. This API is defined in https://github.com/grpc/grpc/blob/master/doc/connectivity-semantics-and-api.md. But I am thinking to get rid of it since it is not very useful and extremely hard to implement properly for a complex picker (e.g., load balancing picker).